### PR TITLE
Set edgecore running as a system service

### PR DESCRIPTION
**Pull request by dingyin**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind test
> /kind failing-test
/kind feature

**What this PR does / why we need it**:
This PR set edgecore running as a system service, set the service auto start after the node reboots.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: keadm/cmd/keadm/app/cmd/util/common.go [hunk 1]
```diff
@@ -425,18 +449,18 @@ func runEdgeCore(version string) error {
...
-		binExec = fmt.Sprintf("%s > %s/%s.log 2>&1 &", KubeEdgeBinaryName, KubeEdgeLogPath, KubeEdgeBinaryName)
-	} else {
-		binExec = fmt.Sprintf("%s > %skubeedge/edge/%s.log 2>&1 &", KubeEdgeBinaryName, KubeEdgePath, KubeEdgeBinaryName)
-	}
-
+	// if version >= "1.1.0" {
+	// 	binExec = fmt.Sprintf("%s > %s/%s.log 2>&1 &", KubeEdgeBinaryName, KubeEdgeLogPath, KubeEdgeBinaryName)
+	// } else {
+	// 	binExec = fmt.Sprintf("%s > %skubeedge/edge/%s.log 2>&1 &", KubeEdgeBinaryName, KubeEdgePath, KubeEdgeBinaryName)
+	// }
```
Comment (kevin-wangzefeng) [inline keadm/cmd/keadm/app/cmd/util/common.go:462]: I guess this would break installation support of pre 1.2 releases?
/cc @fisherxu

**Comment by daixiang0** [comment]
We need to consider the case that edge node does not support systemd, so need to auto detect use systemd service or not.

File: keadm/cmd/keadm/app/cmd/util/common.go [hunk 2]
```diff
@@ -380,6 +381,29 @@ func installKubeEdge(componentType types.ComponentType, arch string, version str
 		}
 	}
 
+	// When installing  edgecore, also download the edgecore.service file from the KubeEdge/build/tools/ and place it in /etc/kubeedge/, if it doesn't already exist.
+	if componentType == types.EdgeCore {
+		edgecoreServiceFileName := "edgecore.service"
+		if _, err = os.Stat(KubeEdgePath + edgecoreServiceFileName); err == nil {
+			fmt.Println("Expected or Default Edgecore service file is already downloaded")
+		} else if !os.IsNotExist(err) {
```
[the remark this answers is not in the document]
Comment (daixiang0) [reply keadm/cmd/keadm/app/cmd/util/common.go:389]: Forget it, sorry.

File: keadm/cmd/keadm/app/cmd/util/common.go [hunk 3]
```diff
@@ -380,6 +381,29 @@ func installKubeEdge(componentType types.ComponentType, arch string, version str
...
+			return err
+		} else {
+			try := 0
+			for ; try < downloadRetryTimes; try++ {
+				cmdStr := fmt.Sprintf("cd %s && sudo wget -k --no-check-certificate %s", KubeEdgePath, EdgecoreServiceFileURL)
```
[the remark this answers is not in the document]
Comment (daixiang0) [reply keadm/cmd/keadm/app/cmd/util/common.go:394]: No need to change this, since we use `wget` to download crd files.

**Comment by daixiang0** [comment]
@dingyin could rebase it?

**Comment by kevin-wangzefeng** [comment]
PR needs rebase

**Pull request closed without merging**